test/zone_test.rb

```diff
@@ -48,6 +48,9 @@ def test_zone_has_config_with_provider
   end
 
   def test_zone_is_invalid_if_provider_does_not_support_record_type
+    supported_record_types = Provider::DynECT.record_types - ['ALIAS']
+    Provider::DynECT.stubs(:record_types).returns(supported_record_types)
```

Comment:
@andrewlouis93 only DynECT zones would be able to specify they accept ALIAS records in [the config block](https://github.com/Shopify/record_store/blob/368153106068800325b4e579483faa427afe7add/template/zones/dnsimple.example.com.yml#L3). You'd have a new possible config along the lines of `supports_alias: {true|false}` (we can set a default of false if missing).